app/models/pull_request.rb

```diff
@@ -159,13 +183,30 @@ def validate(saved_changes)
     end
   end
 
-  private
-
-  ##
-  #  Since we want to be a fancy responsive application we to all the validation
-  #  stuff which might result in some new querys and api requests asyncronously.
-
-  def queue_validation
-    ValidatePullRequestWorker.perform_async(id, saved_changes) if saved_changes? || mergeable.nil?
+  def validate_status
+    label = Label.tests_fail
+    # get current status
+    state = begin
+              statuses = Github.client.combined_status(gh_repository_id, github[:head][:sha])
+              statuses[:state]
+            rescue StandardError
+              nil
```

Comment:
Maybe you want to capture and log the error here and then return. That's probably easier to debug that `state: nil` later on.